aztec/src/main/kotlin/org/wordpress/aztec/AztecTagHandler.kt

```diff
@@ -177,11 +180,23 @@ class AztecTagHandler(val context: Context, val plugins: List<IAztecPlugin> = Ar
     }
 
     private fun start(output: Editable, mark: Any) {
+        tagStack.add(mark)
+
         output.setSpan(mark, output.length, output.length, Spanned.SPAN_MARK_MARK)
     }
 
     private fun end(output: Editable, kind: Class<*>) {
-        val last = output.getLast(kind)
+        // Get most recent tag type from the stack instead of `getLast()`. This is a speed optimization as `getLast()`
+        //  doesn't know that the tags are in fact nested and in pairs (since it's html)
```

Comment:
not sure I'm following correctly, the comment here seems to assume there can't be self-closing tags, for instance?

Reply:
I guess the wording is not the clearest it could be. Self-closing tags (empty elements in html parlance) are treated as pairs of tags by tagsoup (the parsing library we're using) anyway. So for instance, a self-closing tag will cause both a call to the `start()` and the `end()` methods.

Does this address your concern?

I will update the wording to include this detail too.

Reply:
> Self-closing tags (empty elements in html parlance) are treated as pairs of tags by tagsoup 

👌 perfect! thanks :)